lib/auth.ts

```diff
@@ -111,6 +126,81 @@ export class Auth {
     return token;
   };
 
+  /**
+   * Perform a request against Kontist REST API
+   */
+  private request = async (path: string, method: HttpMethod, body?: string) => {
+    if (!this.token) {
+      throw new Error("User unauthorized");
+    }
+
+    const requestUrl = new URL(path, this.baseUrl).href;
+
+    const response = await fetch(requestUrl, {
+      method,
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.token.accessToken}`
+      },
+      body
+    });
+
+    return response.json();
+  };
+
+  /**
+   * Called periodically by `createMFAChallenge` until
+   * the challenge expires or its status is updated
+   */
+  private pollChallengeStatus = (
+    pendingChallenge: Challenge,
+    resolve: Function,
+    reject: Function
+  ) => async () => {
+    const challenge = await this.request(
+      `${MFA_CHALLENGE_PATH}/${pendingChallenge.id}`,
+      HttpMethod.GET
+    );
+
+    if (new Date(challenge.expiresAt) < new Date()) {
+      if (this.challengePollIntervalId) {
```

Comment:
So timeout ids in node have type `NodeJS.Timeout` and in browser they have `number`, so I defined this property like so: `private challengePollIntervalId?: ReturnType<typeof setInterval>;` 

Which works fine but I couldn't find a way to initialise it, if I do `private challengePollIntervalId: ReturnType<typeof setInterval>; = 1234` compiler yells as type `number` is not compatible with type `NodeJS.Timeout`.

So here I have to do these ugly checks 😬any fluent typescripter would have an idea how I can avoid this?

Reply:
you need to have those checks cause clearInterval accepts non-falsy value, right? 
You could also cast it to NodeJS.Timeout like 
```typescript
clearInterval(this.challengePollIntervalId as NodeJS.Timeout);
```
which should be fine for the browser also cause this type won't be in the final bundle file.

Maybe you can also extract it to small function so it won't look that ugly anymore :)

Reply:
> you need to have those checks cause clearInterval accepts non-falsy value, right?

Yep, exactly

> You could also cast it to NodeJS.Timeout like ...

This is a great idea and it works 🙂 I think I will extract a type like this:
`type IntervalID = ReturnType<typeof setInterval>;`
and use it for the property definition `private challengePollIntervalId?: IntervalID;` and cast my timeouts to it `clearInterval(this.challengePollIntervalId as IntervalID);`

Thank you for the suggestion 🙏

Reply:
Have changed it in this commit: ce5eeca110fffd68790f3e6d601736f7b4011a79